### PR TITLE
Add developer invite system

### PR DESCRIPTION
**Pull request by jameswestman**
Adds a system for inviting other developers to a direct upload app.

There is a new Developers tab in the app management page for direct upload apps only. From there, the primary developer can invite or remove other developers, and non-primary developers can leave the app from there as well.

Each user account gets a random, unique invite code displayed on the Developer tab of their account page. To invite a developer, ask them for their code and paste it into the invite dialog. This way, all invites are associated with an existing user account, and we don't have to deal with the UX or backend complexities of creating an account and accepting an invite at the same time.

You must accept the Publisher Agreement to accept an invite.

Emails are sent:
- To the invited user when they receive an invite
- To all developers (including the new one) when an invite is accepted

There is a bug where developers can have a display_name of NULL, and they won't show up in the developers tab. This is probably best solved by overhauling some of the account management code and introducing a "main account" concept.

### CODE REVIEW

File: backend/app/logins.py
Comment (razzeee) [inline backend/app/logins.py:855]: Wondering, if we could generate a word combination instead? `styled hatchet stoke blooms` for e.g. and I guess it could fail, when we insert here?

**Comment by razzeee** [comment]
> There is a bug where developers can have a display_name of NULL, and they won't show up in the developers tab. This is probably best solved by overhauling some of the account management code and introducing a "main account" concept.

Can you create an issue for that?

I would like some backend tests, but I haven't had a look at how complex that part is

File: backend/app/models.py
Comment (razzeee) [inline backend/app/models.py:581]: I'm actually wondering, if we should manage that in a different (but mostly identical) table. As that would decrease the likelihood of giving wrong access when we change stuff up.

It also might allow us, to lock down the table constraints better.

File: frontend/pages/apps/manage/[appId]/accept-invite/index.tsx
Comment (razzeee) [inline frontend/pages/apps/manage/[appId]/accept-invite/index.tsx:110]: My gut feeling is, that this is wrong and likely not needed. We probably don't want to prebuild these pages

File: backend/app/config.py
Comment (razzeee) [inline backend/app/config.py:24]: One thing @barthalion needs to set, when we merge this

File: backend/app/emails.py
Comment (razzeee) [inline backend/app/emails.py:33]: Makes me wonder, if we should send the inviting party an email, when the invitee declines.

File: backend/app/emails.py
Comment (razzeee) [inline backend/app/emails.py:141]: Wondering, if this should be on the model

File: frontend/public/locales/en/common.json
Comment (razzeee) [inline frontend/public/locales/en/common.json:286]:
```suggestion
  "invite-developer-description": "Enter the developers invite code to invite them to become a developer of this app. They can find the code on the Developers tab of their account page.",
```

Comment (razzeee) [reply frontend/public/locales/en/common.json:286]: This also makes me wonder, if we should inject the translation for `Developers` here, to make sure it's not translated differently

File: frontend/pages/apps/manage/[appId]/accept-invite/index.tsx
Comment (jameswestman) [reply frontend/pages/apps/manage/[appId]/accept-invite/index.tsx:110]: What should I use instead? I copied this from the manage page.

Comment (razzeee) [reply frontend/pages/apps/manage/[appId]/accept-invite/index.tsx:110]: I think you should be able to drop the line completely and it should be fine.

File: backend/app/models.py
Comment (jameswestman) [reply backend/app/models.py:581]: Good idea. It might also make it easier to implement ownership transfer.

File: frontend/public/locales/en/common.json
Comment (jameswestman) [reply frontend/public/locales/en/common.json]: I think "developer's" is correct? Fixed the others.

Comment (razzeee) [reply frontend/public/locales/en/common.json]: Reading up on this, I think you're right

File: backend/app/logins.py
Comment (razzeee) [inline backend/app/logins.py:957]: Would be nice to also add this to the type in `Login.ts`

File: backend/app/logins.py
Comment (jameswestman) [reply backend/app/logins.py:855]: Hmm, not sure if the UX is much better, especially for i18n? We should at least split the characters into chunks though.

Comment (razzeee) [reply backend/app/logins.py]: Yeah, that sounds like a neat compromise

File: backend/alembic/versions/c0472125ea51_invite_developers.py
Comment (razzeee) [inline backend/alembic/versions/c0472125ea51_invite_developers.py:25]: I found the inclusion of this field surprising. Is this really used/needed? I feel like that design might be something we would want to talk about. I at least can't come up with a sane one.

I think making somebody to be the primary should be a separate action and happen instantly, without needing the receiving part to accept.

**Review by razzeee — APPROVED** [review]
I think we're almost there, there's just one last thing that I found.

File: backend/alembic/versions/c0472125ea51_invite_developers.py
Comment (jameswestman) [reply backend/alembic/versions/c0472125ea51_invite_developers.py:25]: Yeah, I intended for that to be used to invite someone transfer primary ownership, but didn't implement it yet. It should probably be a separate action, but I'm not sure being able to dump maintainership on someone without their approval is a good idea.

Comment (razzeee) [reply backend/alembic/versions/c0472125ea51_invite_developers.py:25]: Let's remove the field and do it later?

File: frontend/pages/my-flathub.tsx
Comment (razzeee) [inline frontend/pages/my-flathub.tsx:47]: I've started to guard cases like this with is production or is moderator for now.

**Comment by razzeee** [comment]
I'll merge this tomorrow, if nobody speaks up.

**Pull request merged**
